apps/docs/src/components/side-panel/side-panel.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import {useState, useRef} from 'react'
+import {IconButton} from '@primer/react'
+import {Dialog} from '@primer/react/experimental'
+import {ThreeBarsIcon, XIcon} from '@primer/octicons-react'
+import styles from './side-panel.module.css'
+import {Sidebar} from '@primer/gatsby-theme-doctocat'
+
+export function SidePanel() {
+  const [isOpen, setIsOpen] = useState(false)
+  const closeButtonRef = useRef<HTMLButtonElement>(null)
+  const openButtonRef = useRef<HTMLButtonElement>(null)
+
+  const openDialog = () => {
+    setIsOpen(true)
+  }
+
+  const closeDialog = () => {
+    setIsOpen(false)
+  }
+
+  const renderHeader = ({dialogLabelId}: {dialogLabelId: string}) => (
+    <div
+      className={styles.Header}
+      id={dialogLabelId}
+      aria-label="Navigation"
+      role="heading"
+      aria-level={1}
+    >
+      <IconButton
+        ref={closeButtonRef}
+        icon={XIcon}
+        variant="default"
+        aria-label="Close Navigation"
+        onClick={closeDialog}
+      />
+    </div>
+  )
+
+  return (
+    <div className={styles.SidePanel}>
+      <IconButton
+        ref={openButtonRef}
+        icon={ThreeBarsIcon}
+        variant="default"
+        aria-label="Navigation"
+        onClick={openDialog}
+      />
+      {isOpen && (
+        <Dialog
+          onClose={closeDialog}
+          width="small"
+          // TODO: get `@primer/react` updated to a version that supports the `position` prop on experimental `Dialog` component
```

Comment:
Is this something you'll be looking at straight after @mperrotti or shall we take this on in Brand? This workaround feels quite broken with the double scroll bars and centering. 


https://github.com/user-attachments/assets/6c612b02-8c04-446d-b241-9e1947b12bca

Reply:
Double scrollbars should definitely be fixed. I can't think of a way around centering except:
- Upgrade to a newer version of `@primer/react`
- Attempt to use the `sx` prop to fake a side sheet

I'll try using the `sx` prop to manually style this like a side sheet.

Reply:
I ended up achieving this with the `sx` prop

Reply:
Looks and works great on mobile now, thanks @mperrotti